front-end/src/store/analyzeTranscriptSlice.js

```diff
@@ -31,24 +31,28 @@ export const analyzeTranscriptSlice = createSlice({
     setProjectIdToBeDisplayed: (state, action) => {
       state.projectIdToBeDisplayed = action.payload;
     },
-    setBookmarkState: (state, action) => {
+    toggleStar: (state, action) => {
       // get the index of the first intent being displayed
       const firstIndex = state.DisplayingQuestion[state.projectIdToBeDisplayed];
       // get the transcript that is being displayed and has the indices of the intents
       const transcript = state.analyzedTranscripts[state.projectIdToBeDisplayed];
-      // change the bookmark of the intent being changed
+      // change the star state of the intent being changed
       for (let i = firstIndex; i < firstIndex+3; i++) {
-        // should I use if-else or switch?
-        if (transcript[i].question === action.payload.question && action.payload.bookmarkType === "star") {
+        if (transcript[i].question === action.payload) {
           transcript[i].star = !transcript[i].star
-          // or should I make it based off of action payload being passed in from intentDiagram?
-          // where rather than it being based off of whats in state.analyzedTranscripts, it's based off of the API call or the proptype?
-          // transcript[i].star = !action.payload.currentState
         }
-        else if (transcript[i].question === action.payload.question && action.payload.bookmarkType === "flag") {
+      }
+    },
+    toggleFlag: (state, action) => {
+      // get the index of the first intent being displayed
+      const firstIndex = state.DisplayingQuestion[state.projectIdToBeDisplayed];
+      // get the transcript that is being displayed and has the indices of the intents
+      const transcript = state.analyzedTranscripts[state.projectIdToBeDisplayed];
+      // change the bookmark state of the intent being changed
+      for (let i = firstIndex; i < firstIndex+3; i++) {
+        if (transcript[i].question === action.payload) {
           transcript[i].flag = !transcript[i].flag
         }
```

Comment:
Small thing with this for-loop. We should have it exit early if the if-condition executes. Maybe we return true on success and false on failure. When the loop ends we return false. When it succeeds, we short-circuit and return true after changing the attribute?